### ISSUE TITLE
IC2 Uranium Ore and Extreme Reactors Yellorite Ore doesn't spawn at all

### ISSUE THREAD

**Issue by Phoenix1355** [FTBTeam/FTB-Revelation-2#227]
* **Modpack Version**: 3.0.1

* **Issue**: I can't seem to find any uranium or yellorite ore at all, which leads me to believe that it doesn't spawn for some reason. I've tried mining out several large chunks (top to bottom) in the overworld, but still didn't get any uranium or yellorite. This prevents creating anything with Extreme Reactors and makes the mod pack useless, unfortunately.

* **Link to Log or Crash File Paste**: None

* **Is it Repeatable?**: Yes. It doesn't spawn in singleplayer or multiplayer at all.

* **Mod/s Affected**: Industrial Craft 2 + Extreme Reactors

* **Known Fix**: See @poulhp's reply

**Comment by digi-tal-it** [comment]
I can confirm, i haven't found any yellorite or uranium so far and I have been mining quite some time. I was't able to find diamonds as well.

**Comment by thenewjonathan** [comment]
Same for me. Set up a builder and mined a HUGE area with zero yield of anything that would yield refined uranium. Very huge hinderance as this is my main source of reliable energy. Trying to use tree oil and canola oil to get power but it's not working so well. Please fix soon!

**Comment by ghost** [comment]
I performed a similar test, mining out approximately a 40 chunk area at level 22(using draconic staff of power and a /dank/null for quick gathering in single player at 11x11x11) in each of the following dimensions. FTB team, thank you for your continued work on this project and please update as soon as possible.

Diminsions tested: overworld, deep dark, and mining realm.

EDIT: I should also note that I noticed that the level-type=DEFAULT in the server.properties file generated. I also attempted setting it to BIOMESOP and starting a new world, with the same results.

**Comment by poulhp** [comment]
Dont know if it helps anyone but im getting this from the server log:getting this from the server log:  [Error] [CoFH World]: Invalid block name on line 74!. 22:22:08  [Warn] [CoFH World]: Invalid generator for 'uraniumyellorite' on line 72!. 22:22:08  [Error] [CoFH World]: Distribution 'uraniumyellorite' failed to parse its entry on line 72!. 22:22:08  [Error] [CoFH World]: Error parsing populate entry: 'uraniumyellorite' > Please check the parameters..

**Comment by poulhp** [comment]
Dont know if this is a fix or not, but if you remove the line in the cofh overworld config file relating to yellorite ore and then turn on the options in extreme reactors to allow that to spawn the ore insted of cofh it will work.

**Comment by poulhp** [comment]
Pretty sure I got it figured out.
Will update soon, please ignore the post above.

**Comment by poulhp** [comment]
Yep I got it fixed, in the cofh 04_overworld_ores config file.

Change the line that says bigreactors:brore to bigreactors:oreyellorite
and it will spawn once again in your world.

**Comment by ghost** [comment]
@poulhp That appears to have fixed it for me as well. Thanks for the help :)

**Comment by Butterbluemchen12** [comment]
Thanks poulhp! It fixed the problem. But when can we expect an updated version of revelation?

**Comment by Phoenix1355** [comment]
@poulhp Your fix worked for me too, both in singleplayer and multiplayer. Thank you!

**Comment by Butterbluemchen12** [comment]
Ok guys,
i found a new problem relating uranium und yellorium. the quantum quarry isnt able to find any of these - even with the solution of poulhp :(

**Comment by poulhp** [comment]
You are very welcome guys.
Not familier with that quarry?
What mod is that from? Ill have a quick look into it when i get home.

**Comment by Butterbluemchen12** [comment]
Its from extra utilities 2. Thank you very much! :)

**Comment by poulhp** [comment]
it dosnt mine in the overworld? Its like another dimension?

**Comment by Butterbluemchen12** [comment]
ahhh, now it found yellorite. but its by far less than before this update. 
i think it mines in a new dimension.

**Comment by poulhp** [comment]
Is it a complety new world that you made after my fix or did you add the fix and kept playing?

**Comment by Butterbluemchen12** [comment]
I added it and kept my world.

**Comment by poulhp** [comment]
Can you do me a favor and make a new world and test it?

**Comment by Butterbluemchen12** [comment]
yes, ofc give me half an hour :)

**Comment by poulhp** [comment]
Also, please make a backup before doing this. But in your world, not the one you are testing in. Delete the folder called DIM-4. That is the quarry dimension. It might have created that dimension using the old settings before the fix

**Comment by Butterbluemchen12** [comment]
ok, i made a tesworld in creative and made the same setup again.
in my first post I found 4 yellorite ore after my quarry mined 3640 blocks.
In my test world i found even 0 yellorite and uranium ore after 3640 blocks....
did they nerve yellorite and uranium so hard in this version?

**Comment by Butterbluemchen12** [comment]
after 5000 blocks in the testworld i found my first yellorite

**Comment by poulhp** [comment]
Not sure how much spawns since i change the amount in my game. 
But 1 block in 5000 seems way to low. But depending how the quarry mines and how much spawns you can be really unlucky i guess. But i doubt its that. Can you tell me how much its set to spawn per default in cofh config file where you changed the yellorium ore name.?

**Comment by Butterbluemchen12** [comment]
I can show you:
"uraniumyellorite": {
			"distribution": "uniform",
			"generator": {
				"block": [
					{"name": "bigreactors:oreyellorite", "weight":80},
					{"name": "ic2:resource", "metadata": 4, "weight":20}
				],
				"material": "stone",
				"cluster-size": 8
			},
			"cluster-count": 4,
			"min-height": 16,
			"max-height": 42,
			"retrogen": "true",
			"biome": "all",
			"dimension": {
				"restriction": "blacklist",
				"value": [
					-1,
					1
				]
			}

**Comment by poulhp** [comment]
Thank you 😁. 
Okay gonna head on home now. 
First thing ill do is change thoes numbers alot higher to see if the quarry uses the same world gen as the overworld.
Will update when i know more. Thanks for the help

**Comment by poulhp** [comment]
Uhm, do you have the Biome marker item in your mod?

**Comment by poulhp** [comment]
Okay I can confirm that changeing the 
cluster-size": 8
},
"cluster-count": 4,

In the config fil will effect the quarry.
Just did 151 in each of them and that got over 144 yellorium from 360 blockes mined with the quarry.

**Comment by poulhp** [comment]
But that will also change the ammout that spawns in all other diemensions. 
So try to mess around with it and see what feels right.

**Comment by Butterbluemchen12** [comment]
Hey, ok this fixed it! But i have a new thing.. My Quantum Quarry doesnt mine quarried stone although i inserted a biome marker with woodland in it. in normal overworld you can find huge areas of quarried stone in this biome.
In addition to that you cant find biome marker in the crafting search. I goggled the recipe and i was able to build it.

**Comment by poulhp** [comment]
That i can not help you with, the quarry sould be able to mind everything thats in the overworld since it uses that as a template. 
From what iv seen qurried stone is very rare in my world.

**Comment by Butterbluemchen12** [comment]
The quarry seems to be quite bugged. I ran it for about 4 hours on max energy in forest biome for quarried stone and for about 4 hours on ice biome for permafrost stone. it didnt find any of it.
and if i set it on desert biome it doesnt find any sandstone or sand.
Is there any place where i can report these bugs so they can get fixed sometime?

**Comment by rune-san** [comment]
Quantum Quarry is under the [rwtema/Extra-Utilities-2-Source](https://github.com/rwtema/Extra-Utilities-2-Source) repot.

**Comment by thenewjonathan** [comment]
Do we know when the fix will be released?

**Comment by ttlawry** [comment]
> Yep I got it fixed, in the cofh 04_overworld_ores config file.
> Change the line that says bigreactors:brore to bigreactors:oreyellorite
> and it will spawn once again in your world.

Hi, Will this fix work for preexisting worlds? And does this fix work for the mining dimension, as this is the world where all my quarrys are? 

Thanks.

**Comment by ttlawry** [comment]
> Yep I got it fixed, in the cofh 04_overworld_ores config file.
> Change the line that says bigreactors:brore to bigreactors:oreyellorite
> and it will spawn once again in your world.
> 
> Hi, Will this fix work for preexisting worlds? And does this fix work for the mining dimension, as this is the world where all my quarrys are?
> Thanks.

Also does the fix, fix the uranium generation too?

**Comment by poulhp** [comment]
Yes, but only in new chunks, retrogen is set to yes so it might regen it old chunks but you have to enter a new chunk. If its on a server you might have to restart the server to get it to reload the config file once again.
The same applies to the mining dim, it will in new chunk and might it old based on retrogen, but that sometimes work and sometimes it dont, cant quite figurer out how, when any why it works sometimes.
You can delete the mining dim to get it to regen the whole dim but then you'll lose your quarry

Yes the uranium gen is tied to the yellorium. 20% of the yellorium will be uranium if I remember correct.

**Comment by ttlawry** [comment]
Ok thanks very much the for info and the fix :)

**Comment by Jekallz** [comment]
poulhp; when I update my server config files to: 
"uraniumyellorite": {
			"distribution": "uniform",
			"generator": {
				"block": [
					{"name": "bigreactors:oreyellorite", "weight":80},
					{"name": "ic2:resource", "metadata": 4, "weight":20}
				],
				"material": "stone",
				"cluster-size": 151
			},
			"cluster-count": 151,
			"min-height": 16,
			"max-height": 42,
			"retrogen": "true",
			"biome": "all",
			"dimension": {
				"restriction": "blacklist",
				"value": [
					-1,
					1
				]
			}

- Nothing seems to change as far as ore spawning goes. We load new chunks with a 300x300 quarry and still dont find any uranium or yellorite. Is there something I did wrong or what can we do to fix it for the server?

**Comment by Jekallz** [comment]
> Yes, but only in new chunks, retrogen is set to yes so it might regen it old chunks but you have to enter a new chunk. If its on a server you might have to restart the server to get it to reload the config file once again.
> The same applies to the mining dim, it will in new chunk and might it old based on retrogen, but that sometimes work and sometimes it dont, cant quite figurer out how, when any why it works sometimes.
> You can delete the mining dim to get it to regen the whole dim but then you'll lose your quarry
> 
> Yes the uranium gen is tied to the yellorium. 20% of the yellorium will be uranium if I remember correct.

poulhp; when I update my server config files to:
"uraniumyellorite": {
"distribution": "uniform",
"generator": {
"block": [
{"name": "bigreactors:oreyellorite", "weight":80},
{"name": "ic2:resource", "metadata": 4, "weight":20}
],
"material": "stone",
"cluster-size": 151
},
"cluster-count": 151,
"min-height": 16,
"max-height": 42,
"retrogen": "true",
"biome": "all",
"dimension": {
"restriction": "blacklist",
"value": [
-1,
1
]
}

Nothing seems to change as far as ore spawning goes. We load new chunks with a 300x300 quarry and still dont find any uranium or yellorite. Is there something I did wrong or what can we do to fix it for the server?

**Comment by poulhp** [comment]
Server config file? what file is that? and why would you copy that ?

**Comment by Jekallz** [comment]
> Server config file? what file is that? and why would you copy that ?

.... the cofh oregen file for my server. I upped the values for cluster size and count but no change was made as far as oregeneration in new chunks.

**Comment by karikusa** [comment]
It seems that there is a defect in the formation of ore,
Once the configuration file for ore generation
How about reviewing?

**Comment by poulhp** [comment]
Karikusa I have no idea what you mean?

Jekallz well if its in new chunks and (and you are sure its in new chunks) and you restartede the server after you made the change id say that somewhere you did something wrong. Its working for me across 3 servers how and 2 single player games.

I cant really help you since my fix wasnt about copying an entire section into the config but only changing one line.

**Comment by Jekallz** [comment]
I mean all i did was change the values already in the config, I was just showing you the entirety of the uranium part in the cofh worldoregen config file

**Comment by poulhp** [comment]
OKay fair enough. Well then based on what little info I have I would say that you havnt generated any new chunks.
Id need to know more about the quarry you are using.
And you did restart the server after you changed config file right?

**Comment by Jekallz** [comment]
Yea so we are using the Builder from RFtools as our quarry. Even without this tho, we do not find any natural generation in new chunks. We've gone thousands of blocks in the nether, made a portal and gone somewhere unexplored and ungenerated with no luck of finding uranium or yellorite. And yea i restarted the server after changing the config file.

**Comment by poulhp** [comment]
can you give me complete name of the file you changed?

**Comment by poulhp** [comment]
and also post the entire content of that file

**Comment by tfox83** [comment]
This has been fixed in 3.1.0

**Comment by Jekallz** [comment]
Nevermind. The only thing I changed is exactly the same thing you changed in your above posts lol. I didnt do some brand new never seen before thing. You claimed that changing the cluster size and count in the COFH worldoregen file made uranium and yellorite spawn in new chunks. I tried that and that was not the case for me. We're just going to use environmental tech mod to get our uranium since nobody can seem to find a real fix for this obviously broken modpack. Thanks.
